### ISSUE TITLE
eper being excluded from releases

### ISSUE THREAD

**Issue by nickelization** [ninenines/erlang.mk#711]
In current versions of erlang.mk, it looks like the eper package is automatically added to IGNORE_DEPS. I'm not very familiar with the internal workings of erlang.mk, but this does appear to be preventing eper from being included when I build a release. It looks like the intent was probably to exclude tools that are only used in development and testing; however, the redbug tool included in eper is actually intended to be safe for production use, and I have indeed found it useful in the past for diagnosing issues on production systems, so I would like to override this behavior.

If I created a pull request to remove eper from this list, would it be accepted? Other items in the list seem more cut and dried (e.g. eunit_formatters is obviously not useful in a production release) but I'm not sure eper belongs in this list. My understanding is that if it were removed, people would still have the option to specify eper in IGNORE_DEPS in their Makefile, so the current behavior could still be obtained pretty easily...but right now I don't know of any simple way for a project to remove it from the list without directly editing the erlang.mk file.

**Comment by essen** [comment]
Normally eper should be in IGNORE_DEPS only for dependencies detected as Rebar projects, not for your own Erlang.mk project. So if you specify it in DEPS it should be included.

**Comment by nickelization** [comment]
Hmm, okay, perhaps I'm doing something else wrong then, but here's what I'm seeing. I'm working on a project where I've added eper to DEPS, and it does download and build eper, but when I execute "make run", none of the eper beams get copied into `_rel`, and I can't execute redbug commands. But, if I manually edit erlang.mk and remove eper from that IGNORE_DEPS line (under the dep_autopatch_rebar.erl section) then it seems to treat eper like all my other dependencies, everything gets copied into the release, I can use redbug following "make run", etc.

**Comment by essen** [comment]
Can you copy paste the .app file you get after compiling? And if you use a .app.src file, can you confirm you listed eper there? You have to do things manually if you use a .app.src file.

**Comment by nickelization** [comment]
Soooo...I'm really not sure what happened, but I can't seem to reproduce the problem anymore 😅 . I did a fresh clone and build of the project and eper was included properly, then went back and reverted the erlang.mk changes in my original working copy and rebuilt, and in every case eper is getting properly included now. Tried all sorts of combinations of running different clean targets, manually deleting build products, etc., but it always works now.

My guess is that something must have somehow gotten into a bad state yesterday, and my original edit to IGNORE_DEPS had nothing to do directly with fixing the problem, but that maybe editing erlang.mk somehow caused it to rerun some build step that was being skipped...or something. Oh well. Thanks for your help, and sorry to have bothered you! If I ever figure out how to reproduce my original problem then I'll open a new issue, but for now I guess I'll go ahead and close this.
